libnavigation-core/src/test/java/com/mapbox/navigation/core/reroute/MapboxRerouteControllerTest.kt

```diff
@@ -169,6 +179,58 @@ class MapboxRerouteControllerTest {
         }
     }
 
+    @Test
+    fun reroute_success_from_alternative() {
+        mockkStatic(
+            NavigationRoute::routeId,
+            NavigationRoute::routerOrigin,
+            RouteProgress::routeAlternativeId,
+        ) {
+            val routeId1 = "id_1"
+            val routeId2 = "id_2"
+            val mockRoutes = listOf<NavigationRoute>(
+                mockk {
+                    every { routeId } returns routeId1
+                    every {
+                        routerOrigin
+                    } throws IllegalStateException(
+                        "route 1 origin mustn't be invoked: it removes from list and is not mapped"
+                    )
+                },
+                mockk {
+                    every { routeId } returns routeId2
+                    every { routerOrigin } returns com.mapbox.navigator.RouterOrigin.ONLINE
+                },
+            )
+            val expectedRoutes = mockRoutes.drop(1)
+            every { directionsSession.routes } returns mockRoutes
+            every { tripSession.getRouteProgress()?.routeAlternativeId } returns routeId2
+            val slotNewRoutes = slot<List<NavigationRoute>>()
+            val slotOrigin = slot<RouterOrigin>()
+            every {
+                navigationRouteCallback.onNewRoutes(capture(slotNewRoutes), capture(slotOrigin))
+            } just runs
+            addRerouteStateObserver()
+
+            rerouteController.reroute(navigationRouteCallback)
+
+            verify(exactly = 1) {
+                navigationRouteCallback.onNewRoutes(any(), any())
+            }
+            assertEquals(expectedRoutes, slotNewRoutes.captured)
+            assertEquals(RouterOrigin.Offboard, slotOrigin.captured)
```

Comment:
why is route origin `RouterOrigin.Offboard`? We're keep using the same route we get from directions API but selecting a different primary route. Shouldn't it be `RouterOrigin.Onboard`?

Reply:
nah, the SDK sends the origin of the alternative route, which becomes primary. In theory, all routes come from the same origin, so other routes should have the same origin

Reply:
Oh, sorry, I thought `RouterOrigin.Offboard` means offline, but it's vice versa